src/mbedit/mbedit_prog.c

```diff
@@ -3678,7 +3674,11 @@ int mbedit_open_file(char *file, int form, int savemode) {
 		}
 
 		/* handle esf edits */
-		const bool outputmode = output_mode != MBEDIT_OUTPUT_BROWSE;
+    int outputmode;
+    if (output_mode == MBEDIT_OUTPUT_BROWSE)
+      outputmode = MBP_ESF_NOWRITE;
+    else
+      outputmode = MBP_ESF_WRITE;
```

Comment:
Should do this if it's an int:

```c
const int outputmode = output_mode == MBEDIT_OUTPUT_BROWSE ? MBP_ESF_NOWRITE :  MBP_ESF_WRITE;
```

This is really confusing code.  `if (savemode || outputmode)` implies outputmode is a bool.

Reply:
To me any use of a ternary operator is confusing  to read, but I have put it back mostly the way you want.